### PR TITLE
Disable TOC from pages without title, or headings, or with only one.

### LINKED ISSUE #897: Better space management in doc
**Description**
When the page contains no titles and subtitles, TOC (the blank area in orange here) should be disabled to let content (the area in red) expand to the right.

image

### PR DESCRIPTION
**Pull request by jrobinAV**
Resolves #897 

- Incorporating Satoshi's changes #974 
- Generalizing to all documentation chapters.
- Automatic formatting (remove EOL spaces, add EOF line, remove multiple blank lines, etc.)

### CODE REVIEW

File: docs/tutorials/fundamentals/3_complete_application/step_05/step_05.md [hunk 1]
```diff
@@ -105,19 +105,19 @@ performance = Markdown("pages/performance/performance.md")
...
 
-The variables *comparison_scenario*, *metric_selector*, and *selected_metric* are initialized. 
+The variables *comparison_scenario*, *metric_selector*, and *selected_metric* are initialized.
 The *comparison_scenario* DataFrame stores the comparison data, while *metric_selector* holds the options for the metric selector. *selected_metric* is initialized with the default metric, which is the first element of the *metric_selector* list (RMSE).
 
 - **Compare Function**:
 
-The `compare` function takes care of the comparison process. This function is triggered when the user clicks 
-the "Compare" button. It gathers the primary scenarios from the application and then goes through each scenario 
+The `compare` function takes care of the comparison process. This function is triggered when the user clicks
```
Comment (FabienLelaquais) [inline docs/tutorials/fundamentals/3_complete_application/step_05/step_05.md]:
```suggestion
The function *compare()* takes care of the comparison process. This function is triggered when the user clicks
```

File: mkdocs.yml_template [hunk 2]
```diff
@@ -158,12 +158,12 @@ nav:
...
-      - "help": manuals/cli/help.md
-      - "create": manuals/cli/create.md
-      - "run": manuals/cli/run.md
-      - "manage-versions": manuals/cli/manage-versions.md
-      - "version": manuals/cli/version.md
+      - "Command-line interface (CLI)": manuals/cli/index.md
+      - "Help": manuals/cli/help.md
+      - "Create": manuals/cli/create.md
+      - "Run": manuals/cli/run.md
+      - "Manage-versions": manuals/cli/manage-versions.md
```
Comment (FabienLelaquais) [inline mkdocs.yml_template]:
```suggestion
      - "Manage versions": manuals/cli/manage-versions.md
```
??

File: mkdocs.yml_template [hunk 3]
```diff
@@ -207,12 +207,12 @@ nav:
       - "GUI Markdown support": manuals/studio/gui.md
 
     - "Reference":
-      - manuals/refmans/index.md
+      - "About Taipy's Reference Manuals": manuals/refmans/index.md
       - "Python API":
-        - manuals/refmans/reference/index.md
+        - "Taipy Python API packages": manuals/refmans/reference/index.md
```
Comment (FabienLelaquais) [inline mkdocs.yml_template]:
```suggestion
        - "Python APIs": manuals/refmans/reference/index.md
```

File: mkdocs.yml_template [hunk 4]
```diff
@@ -207,12 +207,12 @@ nav:
...
         [REFERENCE_CONTENT]
       - "REST API":
-        - manuals/refmans/reference_rest/index.md
+        - "Taipy's REST APIs": manuals/refmans/reference_rest/index.md
```
Comment (FabienLelaquais) [inline mkdocs.yml_template]:
```suggestion
        - "REST APIs": manuals/refmans/reference_rest/index.md
```

File: docs/tutorials/fundamentals/1_understanding_gui/step_06/step_06.md [hunk 5]
```diff
@@ -5,60 +5,64 @@
...
 
-    ![Layout](images/layout.png){ width=70% : .tp-image-border }
+            tgb.text("Third column")
+        ```
 
+        ![Layout](images/layout.png){ width=70% : .tp-image-border }
 
-- [expandable](../../../../manuals/userman/gui/viselements/standard-and-blocks/expandable.md): creates a block that can
-expand or shrink.
+- [expandable](../../../../manuals/userman/gui/viselements/standard-and-blocks/expandable.md): creates a block that can expand or shrink.
```
Comment (FabienLelaquais) [inline docs/tutorials/fundamentals/1_understanding_gui/step_06/step_06.md]:
```suggestion
- [expandable](../../../../manuals/userman/gui/viselements/generic/expandable.md): creates 
    a block that can expand or shrink.
```

File: docs/tutorials/fundamentals/1_understanding_gui/step_06/step_06.md [hunk 6]
```diff
@@ -5,60 +5,64 @@
 We'll introduce three new helpful controls to enhance the page's visual appeal. These controls are:
 
 - [part](../../../../manuals/userman/gui/viselements/standard-and-blocks/part.md): creates a group of text/visual elements.
```
Comment (FabienLelaquais) [inline docs/tutorials/fundamentals/1_understanding_gui/step_06/step_06.md]:
```suggestion
- [part](../../../../manuals/userman/gui/viselements/generic/part.md): creates a group of text/visual elements.
```

File: docs/tutorials/fundamentals/1_understanding_gui/step_02/step_02.md [hunk 7]
```diff
@@ -13,9 +18,7 @@ pages: [Markdown](../../../../manuals/userman/gui/pages/markdown.md),
 [Python code](../../../../manuals/userman/gui/pages/builder.md). You can choose which you want
 to learn in the code snippets below.
 
-## Visual elements
-
-Taipy uses the **[visual element](../../../../manuals/userman/gui/viselements/index.md)**
+Taipy uses the [visual element](../../../../manuals/userman/gui/viselements/index.md)
 concept to bring interactivity to the application. A visual element is a
 Taipy graphical object displayed on the client. It can be a
 [slider](../../../../manuals/userman/gui/viselements/standard-and-blocks/slider.md), a
```
Comment (FabienLelaquais) [inline docs/tutorials/fundamentals/1_understanding_gui/step_02/step_02.md]:
```suggestion
[slider](../../../../manuals/userman/gui/viselements/generic/slider.md), a
```

**Review by FabienLelaquais — APPROVED** [review]
Minor changes (also for easier merge)

**Review by FlorianJacta — APPROVED** [review]

File: docs/tutorials/fundamentals/3_complete_application/step_05/step_05.md [hunk 8]
```diff
@@ -105,19 +105,19 @@ performance = Markdown("pages/performance/performance.md")
 
 - **Global Variables**:
 
-The variables *comparison_scenario*, *metric_selector*, and *selected_metric* are initialized. 
+The variables *comparison_scenario*, *metric_selector*, and *selected_metric* are initialized.
 The *comparison_scenario* DataFrame stores the comparison data, while *metric_selector* holds the options for the metric selector. *selected_metric* is initialized with the default metric, which is the first element of the *metric_selector* list (RMSE).
```
Comment (jrobinAV) [inline docs/tutorials/fundamentals/3_complete_application/step_05/step_05.md]:
```suggestion
The *comparison_scenario* DataFrame stores the comparison data, while *metric_selector* holds
the options for the metric selector. *selected_metric* is initialized with the default metric,
which is the first element of the *metric_selector* list (RMSE).
```

**Pull request merged**